numba/core/typing/npdatetime.py

```diff
@@ -266,3 +266,14 @@ class DatetimeCmpGt(DatetimeCmpOp):
 @infer_global(operator.ge)
 class DatetimeCmpGE(DatetimeCmpOp):
     key = operator.ge
+
+
+@infer_global(npdatetime_helpers.datetime_minimum)
+@infer_global(npdatetime_helpers.datetime_maximum)
+class DatetimeMinMax(AbstractTemplate):
+    def generic(self, args, kws):
+        assert not kws
+        assert len(args) == 2
+        assert isinstance(args[0], (types.NPDatetime, types.NPTimedelta))
+        assert isinstance(args[1], (types.NPDatetime, types.NPTimedelta))
```

Comment:
This isn't fully accurate because it allows comparison between types.NPDatetime and types.NPTimedelta. I would refactor it as such.

```suggestion
        error_msg = "DatetimeMinMax requires both arguments to be NPDatetime type or both arguments to be NPTimedelta types"
        assert isinstance(args[0], (types.NPDatetime, types.NPTimedelta)), error_msg
        if isinstance(args[0], types.NPDatetime):
            assert isinstance(args[1], types.NPDatetime), error_msg
        else:
            assert isinstance(args[1], types.NPTimedelta), error_msg
```